receiver/azuremonitorreceiver/credentials.go

```diff
@@ -0,0 +1,72 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package azuremonitorreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/azuremonitorreceiver"
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+func loadTokenProvider(host component.Host, idAuth component.ID) (azcore.TokenCredential, error) {
+	authExtension, ok := host.GetExtensions()[idAuth]
+	if !ok {
+		return nil, fmt.Errorf("unknown azureauth extension %q", idAuth.String())
+	}
+	credential, ok := authExtension.(azcore.TokenCredential)
+	if !ok {
+		return nil, fmt.Errorf("extension %q does not implement azcore.TokenCredential", idAuth.String())
+	}
+	return credential, nil
+}
+
+func loadCredentials(logger *zap.Logger, cfg *Config, host component.Host) (azcore.TokenCredential, error) {
+	var cred azcore.TokenCredential
+	var err error
+
+	if cfg.Authentication != nil {
+		logger.Info("'auth.authenticator' will be used to get the token credential")
+		if cred, err = loadTokenProvider(host, cfg.Authentication.AuthenticatorID); err != nil {
+			return nil, err
+		}
+		return cred, nil
+	}
+
+	logger.Warn("'credentials' is deprecated, use 'auth.authenticator' instead")
```

Comment:
Is this deprecation guarded by a `featruegate` somewhere? If not, I would highly recommend as a followup that a featuregate is added to follow the typical deprecation process.

Reply:
The deprecated field is still supported, it is not removed yet

Reply:
That's fine, but if there is a log announcing its deprecation then that usually means there should be some manner of removal plan in place.